tools/traffic/table/blob_metadata.go

```diff
@@ -0,0 +1,71 @@
+package table
+
+// BlobMetadata encapsulates various information about a blob written by the traffic generator.
+type BlobMetadata struct {
+	// key of the blob, set when the blob is initially uploaded.
+	key []byte
+
+	// checksum of the blob.
+	checksum *[16]byte
+
+	// size of the blob, in bytes.
+	size uint
+
+	// blobIndex of the blob.
+	blobIndex uint
+
+	// remainingReadPermits describes the maximum number of remaining reads permitted against this blob.
+	// If -1 then an unlimited number of reads are permitted.
+	remainingReadPermits int
+
+	// index describes the position of this blob within the blobTable.
+	index uint
+}
+
+// NewBlobMetadata creates a new BlobMetadata instance. The readPermits parameter describes the maximum number of
+// remaining reads permitted against this blob. If -1 then an unlimited number of reads are permitted.
+func NewBlobMetadata(
+	key []byte,
+	checksum *[16]byte,
+	size uint,
+	blobIndex uint,
+	readPermits int) *BlobMetadata {
+
+	if readPermits == 0 {
+		panic("readPermits must be greater than 0, or -1 for unlimited reads")
```

Comment:
We could return error instead of crashing when validation fails

Reply:
change made